content/cucumber/step-definitions.md

```diff
@@ -253,3 +253,48 @@ Given("I have {int} {color} balls", function (int1, color) {
 });
 ```
 {{% /block %}}
+
+# Abstract Your Steps
+
+It is always a good idea to make sure that you are not writing duplicate/similar step definitions. While there are a number of ways to do so and you can document your step definitions as well, it is always good to make use of **helper methods** to abstract your step definitions and reduce the number of steps.
+
+For example, if you have the following steps used across multiple scenarios :
+
+```
+    Given I go to the home page
+    Given I check the about page of the website 
+    Given I get the contact details
+```
+
+and all of these steps have different step definitions which visit the Home, About and Contact pages, you might be writing _redundant steps_. It is possible that the underlying code for executing each of these test cases is different, **but** it is important to understand that from the point of view of their **behavior**, all of these steps do essentially similar things, ie.
+
+_* Open the Home page_
+_* Open the About page_
+_* Open the Contact page_
+
+As such, you can use abstract helper methods to reduce all these into one step :
+
+    Given I go to the {} page
+
+with the step def :
+
+```java
+@Given("I want to open the {string} page")
+public void i_want_to_open_page(String name) {
+  pageFactory.openPage(name);
+}
+```
+
+And then inside your step definition, use a helper method to determine how the page must be opened and direct your step definition to the correct code for opening that particular page.
+
+This helps you in a number of ways like,
+
+1. Allowing you to have less and more easily maintainable steps.
+2. Making your project easily scalable: Adding tests for a new functionality with the same underlying _behavior_ is a lot easier and less cumbersome.
+3. Making your test cases concise and easy to understand by anyone.
+
+to name a few.
+
+You can use the same method to write steps for validating a webpage, clicking a button, etc each with their own helper methods.
+
+A good look at the _(Abstract) Factory Design_ pattern can be very useful for creating/refractoring your test cases with such abstraction. Also, using [Data Tables](/cucumber/api/#data-tables) for providing inputs along with this method makes your steps even more easy to maintain and understand.
```

Comment:
A link to a good explanation of this pattern might be helpful.

Reply:
Will do. I was actually thinking about it but wasn't sure if thats allowed.

Reply:
We can link to external sites, but prefer to do so "sparingly". If you'd like some more guidelines check the [contributing to documentation](https://docs.cucumber.io/community/contributing-to-documentation/) page. Please let me know if anything is unclear!

Reply:
Thank You :) I ll have a look. For now I have added a link